### ISSUE TITLE
`fit-content()` `Formal syntax` section shows an error

### ISSUE THREAD

**Issue by Hexstream** [mdn/content#37441]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function

### What specific section or headline is this issue about?

Formal syntax

### What information was incorrect, unhelpful, or incomplete?

`Error: could not find syntax for this item`

### What did you expect to see?

The formal syntax for `fit-content()`.

### Do you have any supporting links, references, or citations?

Apparently, [this should be treated the same as `minmax()`](https://github.com/orgs/mdn/discussions/247#discussioncomment-3793169), [which does have a working `Formal syntax` section](https://developer.mozilla.org/en-US/docs/Web/CSS/minmax#formal_syntax).

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/fit-content_function`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/fit-content_function/index.md
* Last commit: https://github.com/mdn/content/commit/c9f96f06d4fbd265808f298eb9b2773f739860c5
* Document last modified: 2024-12-20T11:09:30.000Z

</details>

**Comment by wbamberg** [comment]
Thank for filing!

> Apparently, [this should be treated the same as `minmax()`](https://github.com/orgs/mdn/discussions/247#discussioncomment-3793169), [which does have a working `Formal syntax` section](https://developer.mozilla.org/en-US/docs/Web/CSS/minmax#formal_syntax).

"treated the same as `minmax()`", when this comment was written, meant not including formal syntax, because neither of those functions were marked up as functions in the specs, so didn't appear in the [@webref/css](www.npmjs.com/package/@webref/css) package, which is used as the source for this content.

But recent PRs like https://github.com/mdn/content/pull/37210 then added formal syntax for all functions, including `fit-content()`. What I don't understand though is why this works in the preview page (https://pr37210.content.dev.mdn.mozit.cloud/en-US/docs/Web/CSS/fit-content_function#formal_syntax) but not in production. Perhaps they are using different versions of @webref/css?

**Comment by skyclouds2001** [comment]
yes, it seems very strange...

**Comment by OnkarRuikar** [comment]
We have two pages for `fit-content`:
1. https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content - spec [Box sizing module](https://drafts.csswg.org/css-sizing-4/#valdef-width-fit-content)
2. https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function - wrongly shows box sizing module as specs but should be shown as part of [the grid layout module](https://drafts.csswg.org/css-grid/#funcdef-grid-template-columns-fit-content) (ping @estelle). And[ webref does have syntax for it](https://github.com/w3c/webref/blob/6156b406e9f6298333ff8fb8486e4fff6bb010e4/ed/css/css-grid.json#L88-L91).

***

In the local dev environment, `yarn start` (Yari) shows the syntax, and `yarn start:rari` doesn't. (ping @fiji-flo )
```logs
[server] buildDocumentFromURL(/en-US/docs/Web/CSS/fit-content_function): 483.446ms
[server] using http://localhost:8083/en-US/docs/Web/CSS/fit-content_function
[rari]  WARN serve{req=11}:url{/en-US/docs/Web/CSS/fit-content_function}:page{en-US:Web/CSS/fit-content_function}
  :templ{templ="csssyntax" line=49 col=1}
  : rari_doc::templ::render: Error: could not find syntax for this item
```
 Prod has moved to Rari, but GitHub workflows are still using Yari. Based on the discussions about other PRs, GitHub workflows will be updated in the near future.

**Comment by wbamberg** [comment]
> We have two pages for `fit-content`:
> 
> 1. https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content - spec [Box sizing module](https://drafts.csswg.org/css-sizing-4/#valdef-width-fit-content)
> 
> 2. https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function - wrongly shows box sizing module as specs but should be shown as part of [the grid layout module](https://drafts.csswg.org/css-grid/#funcdef-grid-template-columns-fit-content) (ping @estelle). And[ webref does have syntax for it](https://github.com/w3c/webref/blob/6156b406e9f6298333ff8fb8486e4fff6bb010e4/ed/css/css-grid.json#L88-L91).

Yes, I know there are two pages for `fit-content`. This issue about the page for the `fit-content()` function. We don't include formal syntax in keyword pages.

> Prod has moved to Rari, but GitHub workflows are still using Yari. Based on the discussions about other PRs, GitHub workflows will be updated in the near future.

Ah, OK, so this seems like a rari issue.

**Comment by fiji-flo** [comment]
Sorry for this. There was no occurrence of `{{csssyntax}}` using an argument. I'll release support for this nowish.

I see this being used 3 times:

- `font-stretch`: an alias to `font-width` this has been around for a while but worked since cssref contained `font-stretch` if I see this correct.
- `fit-content`: stripping the `_function` from the slug.
- `url`: stripping the `_function` from the slug.

It's easy to strip and option `_function` for page-type `css-function` in the macro if we want this. Then there's only the aliasing case left. Any idea?

(I might respond slowly as I'm only back working next week)

**Comment by skyclouds2001** [comment]
for references:

https://github.com/mdn/content/pull/37212
https://github.com/mdn/content/pull/37211
https://github.com/mdn/content/pull/37210

**Comment by OnkarRuikar** [comment]
>  This issue is about the page for the `fit-content()` function. We don't include formal syntax in keyword pages.

:confused: You called it a function yourself! Unlike `fit-content`, the `fit-content()` is a [value function](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions). How is `grid-template-rows: fit-content(40%)` different from `width: calc(2em * 5)` in terms of being a value function? Or `width: fit-content(10%);` different from `width: var(--default-width)`?

***

> Then there's only the aliasing case left. Any idea?

- I've raised https://github.com/w3c/webref/issues/1427

**Comment by estelle** [comment]
> > 2. https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content_function - wrongly shows box sizing module as specs but should be shown as part of [the grid layout module](https://drafts.csswg.org/css-grid/#funcdef-grid-template-columns-fit-content) (ping @estelle). And[ webref does have syntax for it](https://github.com/w3c/webref/blob/6156b406e9f6298333ff8fb8486e4fff6bb010e4/ed/css/css-grid.json#L88-L91).

There definitely needs to be a "fit-content()" function page. 
We document all CSS functions but not all keywords, so this page can be considered "required" for MDN completeness.
While a `fit-content` keyword page is not "required" by this definition, it is a complex topic and is definitely needed.

**Comment by caugner** [comment]
The error was fixed in [rari 0.1.13](https://github.com/mdn/rari/releases/tag/v0.1.13).

**Comment by wbamberg** [comment]
> > This issue is about the page for the `fit-content()` function. We don't include formal syntax in keyword pages.
> 
> 😕 You called it a function yourself! Unlike `fit-content`, the `fit-content()` is a [value function](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions). How is `grid-template-rows: fit-content(40%)` different from `width: calc(2em * 5)` in terms of being a value function? Or `width: fit-content(10%);` different from `width: var(--default-width)`?
> 

I don't understand the point you are trying to make. I called it a function because it is, and I don't think anyone here has ever said it wasn't. And yes, there is also a `fit-content` keyword, but that doesn't seem at all relevant to the issue.

Anyway, this is fixed now, thank you @fiji-flo !
